week-4/C-methods-this-keyword/README.md

```diff
@@ -0,0 +1,19 @@
+### Goals
+- make objects even more useful by introducing object methods
+- introduce the `this` keyword **without going into any details about how it works** other than the fact that its a way to access the current object's properties/methods

```

Comment:
I'm not keen on introducing `this` because:
a) can achieve the same with simpler solutions
b) introducing it in this way creates a false understanding that `this` references the object literal 
c) `this` is overused anyway. IMO only use it when you there's not other solution

Example without using `this`:

```js
var athlete = {
    name: 'Usain Bolt',
    thirstLevel: 5,
    drink: function() {
        athlete.thirstLevel = athlete.thirstLevel + 2;
    }
};
```

This solution has the advantage of always working as expected whereas use of `this` can lead to unexpected results depending on context.

Reply:
Yep - that looks like a good alternative. I'll use that, thanks.

With this topic I mainly wanted to make objects a bit more versatile by introducing methods to make them do stuff, give them some "life" and similarities to real-world objects, hence the "athlete" example.

I agree with points b) and c). However I don't think it would be that easy for them to get unexpected results with `this`. They'd have to either have other objects "borrow" the `drink` function, or `bind/call/apply` on it - both very unlikely scenarios (please correct me if I've missed anything).

Reply:
I think it's ok to have `this` without knowing full implications. They will also be using `this` often in React and the scope issues are less relevant with es6.